store/views.py

```diff
@@ -12,58 +15,110 @@ def index(request):
+        else:
+            rate = 0.0
+    else:
+        rate = 0.0    
+    if num==0:
+        context = {
+            'book': book, # set this to an instance of the required book
+            'num_available': 'Sorry Book not available', # set this to the number of copies of the book available, or 0 if the book isn't available
+            'rate': rate,
+        }
+    else:
+        context = {
+            'book': book,
+            'num_available': num,
+            'rate': rate,
+        }
+    # START YOUR CODE HERE
     
     return render(request, template_name, context=context)
 
 
 @csrf_exempt
 def bookListView(request):
     template_name = 'store/book_list.html'
+
+    title = request.GET.get("title")
+    author = request.GET.get("author")
+    genre = request.GET.get("genre")
+    if(title and author and genre):
+        books = Book.objects.filter(title=title,author=author,genre=genre)
+    elif(title and author):
+        books = Book.objects.filter(title=title,author=author)
+    elif(title and genre):
+        books = Book.objects.filter(title=title,genre=genre)
+    elif(author and genre):
+        books = Book.objects.filter(author=author,genre=genre)
+    elif(title):
+        books = Book.objects.filter(title=title)
+    elif(author):
+        books = Book.objects.filter(author=author)
+    elif(genre):
+        books = Book.objects.filter(genre=genre)
+    else:
+        books = Book.objects.all()
+
     context = {
-        'books': None, # set this to the list of required books upon filtering using the GET parameters
+        'books': books, # set this to the list of required books upon filtering using the GET parameters
                        # (i.e. the book search feature will also be implemented in this view)
     }
-    get_data = request.GET
-    # START YOUR CODE HERE
-    
-    
+
     return render(request, template_name, context=context)
 
 @login_required
 def viewLoanedBooks(request):
     template_name = 'store/loaned_books.html'
-    context = {
-        'books': None,
-    }
     '''
     The above key 'books' in the context dictionary should contain a list of instances of the 
     BookCopy model. Only those book copies should be included which have been loaned by the user.
     '''
     # START YOUR CODE HERE
-    
-
+    user = User.objects.get(username = request.user.username)
+    books = BookCopy.objects.filter(borrower = user)
+    context = {
+        'books': books,
+    }
 
     return render(request, template_name, context=context)
 
 @csrf_exempt
 @login_required
 def loanBookView(request):
-    response_data = {
-        'message': None,
-    }
     '''
     Check if an instance of the asked book is available.
     If yes, then set the message to 'success', otherwise 'failure'
     '''
     # START YOUR CODE HERE
-    book_id = None # get the book id from post data
-
+    book_id = request.POST.get("bid") # get the book id from post data
+    book = Book.objects.get(id=book_id)
```

Comment:
This may fail with invalid book ID given in POST request, and would lead to server error. Expected behavior is to inform user with Not found (404) error.